packages/@lwc/synthetic-shadow/src/faux-shadow/no-patch-utils.ts

```diff
@@ -0,0 +1,41 @@
+import { isHostElement } from './shadow-root';
+import { getAllMatches, getNodeOwner, getAllSlottedMatches } from './traverse';
+import { ArrayFilter, ArraySlice, isNull, isUndefined } from '@lwc/shared';
+import { getNodeKey, getNodeOwnerKey } from './node';
+import { isGlobalPatchingSkipped } from '../shared/utils';
+
+export function getNonPatchedFilteredCollectionResult(context: Element, nodeList): Element[] {
```

Comment:
This is the logic to filter a partial result of `getElementsByTagName`, `getElementsByClassName` and `getElementsByTagNameNS`